scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SIG # to the trap that catches exit SIG'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Nix is pure, and claims to be transactional.
+    # So show people clean transactional installation.
+
+    createDest() {
+
+        trap destCreateRevert 1
+
+        destCreateRevert() {
+            # all means failed, Exit!
+            error "
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
+
+        # Trying to create $dest by all means
+
+        # If user has rights to successfully execute $cmd. Else -> try to elevate rights section
+        if sh -c "$cmd" > /dev/null 2>&1 ; then
+            print "Creating $dest directory by running $cmd using $USER rights"
+        else
+            if command -v sudo > /dev/null 2>&1 ; then
+                print "Creating $dest directory by running $cmd using sudo:"
+                # Using sudo to create directory
+                if ! sudo sh -c "$cmd" && print "Exiting privileged mode." ; then
+                    >&2 notice "
+
+    Could not succefully execute command with sudo.
+    "
+                fi
+            else
+                print "'sudo' seems not to be installed"
+            fi
 
-# Install an SSL certificate bundle.
-if [ -z "$NIX_SSL_CERT_FILE" ] || ! [ -f "$NIX_SSL_CERT_FILE" ]; then
-    $nix/bin/nix-env -i "$cacert"
-    export NIX_SSL_CERT_FILE="$HOME/.nix-profile/etc/ssl/certs/ca-bundle.crt"
-fi
+        fi
 
-# Subscribe the user to the Nixpkgs channel and fetch it.
-if ! $nix/bin/nix-channel --list | grep -q "^nixpkgs "; then
-    $nix/bin/nix-channel --add https://nixos.org/channels/nixpkgs-unstable
-fi
-if [ -z "$_NIX_INSTALLER_TEST" ]; then
-    $nix/bin/nix-channel --update nixpkgs
-fi
+        # If dest still not created (with user or sudo) - use su to access root
+        if [ ! -d "$dest" ] ; then
+            print "Trying to get privileges using su and root password"
+            if ! su root sh -c "$cmd" && print "Exiting priviliged mode."; then
+                exit 1 # Look into the trap
+            fi
+        fi
+    }
+
+    createStore() {
+        if ! mkdir -p "$dest"/store; then
+            error "
+
+    Script tried to create $dest/store , but it already exists.
+    We do checks on $dest/store before installation.
+    Probaly something went wrong with the installation script.
 
-added=
-if [ -z "$NIX_INSTALLER_NO_MODIFY_PROFILE" ]; then
+    Please:
+      * gather information of your setup conditions
+      * search does BUG exists
```

Comment:
determine whether the bug already exists in location X

Reply:
Yes. Probably better to find a way to wrap bug report message in a function.
Inside error message I probably would make a check for making a bug report message.
Or make a separate function.

`error` for now exits. `errorRevert` relies on a `trap`.
It would be good to be able to post bug report message directly after both of them.

Reply:
Yes, I would cover this minor thing if it is ever got merged.